src/restconfig/src/main/java/org/geoserver/rest/security/UrlCheckController.java

```diff
@@ -0,0 +1,206 @@
+            MediaType.APPLICATION_XML_VALUE,
+            MediaType.TEXT_HTML_VALUE
+        })
+public class UrlCheckController extends RestBaseController {
+
+    private static final Logger LOGGER = Logging.getLogger(UrlCheckController.class);
+
+    private final URLCheckDAO urlCheckDao;
+
+    @Autowired
+    public UrlCheckController(URLCheckDAO urlCheckDAO) {
+        this.urlCheckDao = urlCheckDAO;
+    }
+
+    @GetMapping
+    public RestWrapper<AbstractURLCheck> urlChecksGet() throws IOException {
+        List<AbstractURLCheck> checks = urlCheckDao.getChecks();
+        return wrapList(checks, AbstractURLCheck.class);
+    }
+
+    @GetMapping("/{urlCheckName}")
+    public RestWrapper<AbstractURLCheck> urlCheckGet(@PathVariable String urlCheckName)
+            throws IOException {
+
+        AbstractURLCheck check = urlCheckDao.getCheckByName(urlCheckName);
+        if (check == null) {
+            throw new ResourceNotFoundException("No such URL check found: '" + urlCheckName + "'");
+        }
+
+        return wrapObject(check, AbstractURLCheck.class);
+    }
+
+    @PostMapping(
+            consumes = {
+                MediaType.TEXT_XML_VALUE,
+                MediaType.APPLICATION_XML_VALUE,
+                MediaTypeExtensions.TEXT_JSON_VALUE,
+                MediaType.APPLICATION_JSON_VALUE
+            })
+    @ResponseStatus(HttpStatus.CREATED)
+    public ResponseEntity<String> urlCheckPost(
+            @RequestBody AbstractURLCheck urlCheck, UriComponentsBuilder builder) {
+        try {
+
+            if (urlCheckDao.getCheckByName(urlCheck.getName()) != null) {
+                throw new RestException(
+                        "URL check '" + urlCheck.getName() + "' already exists",
+                        HttpStatus.CONFLICT);
+            }
+
+            verifyCheckName(urlCheck);
+            verifyCheckConfiguration(urlCheck);
+
+            /* enforce URL check enabled status by default */
+            if (urlCheck.isEnabled() == null) {
+                urlCheck.setEnabled(true);
+            }
+            urlCheckDao.save(urlCheck);
+
+            String name = urlCheck.getName();
+
+            LOGGER.log(Level.FINEST, "Added urlCheck {0}", name);
+
+            return new ResponseEntity<>(
+                    name, composeResponseHeaders(builder, name), HttpStatus.CREATED);
+
+        } catch (IOException ex) {
+            throw new RestException(
+                    "Error occurred in creating a new URL check",
+                    HttpStatus.INTERNAL_SERVER_ERROR,
+                    ex);
+        }
+    }
+
+    private static void verifyCheckName(AbstractURLCheck urlCheck) {
+        String checkName = urlCheck.getName();
+        if (checkName == null || checkName.isEmpty()) {
+            throw new RestException("The URL check name is required", HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    private HttpHeaders composeResponseHeaders(UriComponentsBuilder builder, String name) {
```

Comment:
This is not a generic composition like the name seems to suggest, it's for a 201, created.

Reply:
Thanks! 
Renamed.